libs/execution_context/ebpf_native.c

```diff
@@ -1187,9 +1184,11 @@ _ebpf_native_resolve_helpers_for_program(
         goto Done;
     }
 
-    // Update the addresses in the helper entries.
-    for (uint16_t i = 0; i < helper_count; i++) {
-        helpers[i].address = helper_addresses[i];
+    if (helper_count > 0) {
```

Comment:
This if test is not necessary since line 1189 will already do the check as part of the i<helper_count initial check.
Hence can revert the change to this block to keep code complexity lower.